test/compiler/inline.jl

```diff
@@ -51,6 +51,12 @@ function test_outer(a)
 end
 test_inlined_symbols(test_outer, Tuple{Int64})
 
+# Test case 1 for anonymous functions (issue #34939)
+function test_outer2(x)
+    @inline x -> x^2 + 2x - 1
+end
+test_inlined_symbols(test_outer2, Tuple{Int64})
```

Comment:
This doesn't test that the inline declaration has been applied to the inner function; it tests that the inlining pass produced valid output for `test_outer2`.

Reply:
@JeffBezanson 
that makes sense. How do I test if the inline declaration has been applied to the function?
Do I have to look at the `Expr` produced when using the inline macro?

```
julia> Base.parse_input_line("x -> x^2 + 2x - 1")
:($(Expr(:toplevel, :(#= none:1 =#), :((x->begin
          #= none:1 =#
          (x ^ 2 + 2x) - 1
      end)))))

julia> Base.parse_input_line("@inline x -> x^2 + 2x - 1")
:($(Expr(:toplevel, :(#= none:1 =#), :(#= none:1 =# @inline((x->begin
              #= none:1 =#
              (x ^ 2 + 2x) - 1
          end))))))

```

Something like this?

Reply:
This is an easy way to test whether the inlining flag is set:
```
julia> ccall(:jl_ast_flag_inlineable, Bool, (Any,), first(methods(x->x)).source)
false
```

Reply:
thanks! I added some tests to check `:jl_ast_flag_inlineable` for both anonymous and normal functions in b8ddf2a

Reply:
This test can be removed.

Reply:
removed in 57ecb07